client/app/components/ui/modal.hbs

```diff
@@ -0,0 +1,49 @@
+{{#if @open}}
+  <EmberWormhole @to={{if @container @container "reveal-modal-container"}}>
```

Comment:
I don't see `@container` defined anywhere. Is this just so the `@to` can be optionally passed? Would we ever need to wormhole the modal to a different place in the markup? I think the layout will only work if the modal is rendered in that one place, z-indexed after everything in the main application template.

Reply:
Regardless, I love that you've moved the wormhole from the content template to this component. ❤️

Reply:
@andycochran hmm, this code makes 'reveal-modal-container' the default target. This is a design choice. Basically, I want the modal to make no assumptions about outside context. Having the default sort of goes against that, but this kind of approach makes it easier, especially in testing, to create whatever outside context you need. It's really a matter of object-oriented design, specifically encapsulation.

Reply:
Cool. If it helps testing, great. But I can't think of a situation where a modal would need a different target; it must be added to the DOM in one specific place in the markup.

Reply:
@andycochran I get that, but the point is to design this as if it doesn't know anything about the outside markup.

In fact, we should have this paradigm in mind, always: design components to be unaware of outside context. The more we do that, the easier it is to maintain our code.

I get that you can't think of a situation now, but maybe there is one later? I don't know. I think if I follow the comment you make below, this'll be clearer, but I hope the principle makes sense.